lib/manager/nuget/extract.ts

```diff
@@ -1,19 +1,65 @@
+import findUp from 'find-up';
+import * as path from 'path';
+import { XmlDocument } from 'xmldoc';
+import { readFile } from 'fs-extra';
 import { logger } from '../../logger';
 import { get } from '../../versioning';
 import { PackageDependency, ExtractConfig, PackageFile } from '../common';
 import * as semverVersioning from '../../versioning/semver';
 import * as datasourceNuget from '../../datasource/nuget';
 import { SkipReason } from '../../types';
 
-export function extractPackageFile(
+async function determineRegistryUrls(
+  packageFile: string,
+  localDir: string
+): Promise<string[]> {
+  const nuGetConfigPath = await findUp('NuGet.config', {
+    cwd: path.dirname(path.join(localDir, packageFile)),
+    type: 'file',
+  });
+  if (nuGetConfigPath && nuGetConfigPath.startsWith(localDir)) {
+    logger.debug(`found NuGet.config at '${nuGetConfigPath}'`);
+    const nuGetConfig = new XmlDocument(
```

Comment:
Does this ever throw an error?

Reply:
I think we need a global try catch

Reply:
Done in 30346345cc023d5ae57c12c3930ac630a515eac4